python/ray/serve/router.py

```diff
@@ -161,8 +167,33 @@ def _all_query_refs(self):
     def _drain_completed_object_refs(self) -> int:
         refs = self._all_query_refs
         done, _ = ray.wait(refs, num_returns=len(refs), timeout=0)
-        for replica_in_flight_queries in self.in_flight_queries.values():
-            replica_in_flight_queries.difference_update(done)
+        replicas_to_remove = []
+        for replica_info, replica_in_flight_queries in self.in_flight_queries.items():
+            completed_queries = replica_in_flight_queries.intersection(done)
+            if len(completed_queries):
+                try:
+                    ray.get(list(completed_queries))
```

Comment:
@simon-mo  does this means that we'll retrieve the result to the DAG driver? I think it's going to introduce a lot of overhead right?